executor/analyze.go

```diff
@@ -897,8 +898,33 @@ func (e *AnalyzeFastExec) handleSampTasks(bo *tikv.Backoffer, workID int, err *e
 }
 
 func (e *AnalyzeFastExec) buildHist(ID int64, collector *statistics.SampleCollector, tp *types.FieldType) (*statistics.Histogram, error) {
-	// TODO: build histogram and cmsketch here for one collector.
-	return nil, nil
+	// build collector properties.
+	collector.UpdateTotalSize()
+	collector.Samples = collector.Samples[:e.sampCursor]
+	collector.Count = int64(e.sampCursor)
+	data := make([][]byte, 0, len(collector.Samples))
+	for _, sample := range collector.Samples {
+		bytes, err := sample.Value.ToString()
```

Comment:
Why `ToString`? It should be the same as stored bytes.

Reply:
The `GetBytes` only gets the bytes field of `Datum`. It's not the value of `Datum`.

Reply:
Yes, I mean you should use `encode` function to get the data, not the string representation.

Reply:
What the means of `encode` a `Datum`......

Reply:
How about using the `ToBytes`?

Reply:
You can use `ConvertTo` to convert it to bytes.